com.microsoft.mrtk.input/Visualizers/ControllerVisualizer/ControllerVisualizer.cs

```diff
@@ -125,11 +131,12 @@ private void RenderControllerVisuals(UnityInputSystem.InputDevice inputDevice)
         private static readonly Quaternion controllerModelRotatedOffset = Quaternion.Euler(0, 180, 0);
 
         /// <summary>
-        /// Tries to instantiate controller visuals for the specified hand node.
+        /// Tries to instantiate controller visuals for the specified input device
         /// </summary>
+        /// <param name="inputDevice">The input device we want to generate visuals for</param>
         /// <param name="usePlatformVisuals">Whether or not to try to load visuals from the platform provider</param>
         /// <param name="useFallbackVisuals">Whether or not to use the fallback controller visuals</param>
-        private async void InstantiateControllerVisuals(bool usePlatformVisuals, bool useFallbackVisuals)
+        private async void InstantiateControllerVisuals(UnityInputSystem.InputDevice inputDevice, bool usePlatformVisuals, bool useFallbackVisuals)
```

Comment:
Is there going to be any weird edge case here for hands that are not actually input devices (like Quest/Leap hands)? Sure, they don't need controller models, but will this method get called for them ever?

Reply:
Do Quest/Leap hands not register a "tracked" action? I had a bug previously on Quest/Leap where the fallback models would show up when using handtracking, so I thought that they did at least raise actions which had an associated input device.

Regardless, this shouldn't be a problem. The visualizer only tries to instantiate visuals for controllers that raise a "tracked" input action, which should have an associated input device.

Reply:
I don't....think... so? Actually @keveleigh might have added it to the Plugin recently (a basic input device spun up for hands that basically just have a tracked/yes/no value)

Reply:
Ah yeah, we should get a "tracked" action from the articulated hand data becoming active, as Finn described